src/Symfony/Component/Cache/Adapter/AbstractAdapter.php

```diff
@@ -46,9 +46,17 @@ protected function __construct(string $namespace = '', int $defaultLifetime = 0)
             function ($key, $value, $isHit) use ($defaultLifetime) {
                 $item = new CacheItem();
                 $item->key = $key;
-                $item->value = $value;
+                $item->value = $v = $value;
                 $item->isHit = $isHit;
                 $item->defaultLifetime = $defaultLifetime;
+                // Detect wrapped values that encode for their expiry and creation duration
+                // For compacity, these values are packed in the key of an array using magic numbers
+                if (\is_array($v) && 1 === \count($v) && 10 === \strlen($k = \key($v)) && "\x9D" === $k[0] && "\0" === $k[5] && "\x5F" === $k[9]) {
```

Comment:
Magic 10 here, not sure if we want to make it explicit (as we would want to make 5 and 9 explicit as well).